controllers/iamauthpolicy_controller.go

```diff
@@ -249,13 +275,50 @@ func (c IAMAuthPolicyController) updatePolicyCondition(ctx context.Context, k8sP
 		status = metav1.ConditionFalse
 	}
 	cnd := metav1.Condition{
-		Type:               string(gwv1alpha2.PolicyConditionAccepted),
-		Status:             status,
-		ObservedGeneration: k8sPolicy.Generation,
-		LastTransitionTime: metav1.Now(),
-		Reason:             string(reason),
+		Type:   string(gwv1alpha2.PolicyConditionAccepted),
+		Status: status,
+		Reason: string(reason),
 	}
 	meta.SetStatusCondition(&k8sPolicy.Status.Conditions, cnd)
 	err := c.client.Status().Update(ctx, k8sPolicy)
 	return err
 }
+
+func iamAuthPolicyMapFunc(c client.Client, log gwlog.Logger) handler.MapFunc {
+	return func(obj client.Object) []ctrl.Request {
+		requests := []ctrl.Request{}
+		policies := &anv1alpha1.IAMAuthPolicyList{}
+		err := c.List(context.Background(), policies, &client.ListOptions{Namespace: obj.GetNamespace()})
+		if err != nil {
+			log.Error(err)
+			return requests
+		}
+		for _, policy := range policies.Items {
+			if obj.GetName() == string(policy.Spec.TargetRef.Name) {
```

Comment:
Do you need to check `objKind == policy.Spec.TargetRef.Kind` ?

https://github.com/aws/aws-application-networking-k8s/blob/a2d5b973b3516f11784d1a764830e8de2eadf4a7/controllers/accesslogpolicy_controller.go#L346-L349

Reply:
when we create create a "watcher" we already declare which kind of resource to watch, this condition is redundant